x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_exploration/components/classification_exploration/results_table.tsx

```diff
@@ -94,307 +62,65 @@ interface Props {
 
 export const ResultsTable: FC<Props> = React.memo(
   ({ indexPattern, jobConfig, jobStatus, setEvaluateSearchQuery }) => {
-    const [pageIndex, setPageIndex] = useState(0);
-    const [pageSize, setPageSize] = useState(25);
-    const [selectedFields, setSelectedFields] = useState([] as Field[]);
-    const [docFields, setDocFields] = useState([] as Field[]);
-    const [depVarType, setDepVarType] = useState<ES_FIELD_TYPES | undefined>(undefined);
-    const [isColumnsPopoverVisible, setColumnsPopoverVisible] = useState(false);
-    const [searchQuery, setSearchQuery] = useState<SavedSearchQuery>(defaultSearchQuery);
-    const [searchError, setSearchError] = useState<any>(undefined);
-    const [searchString, setSearchString] = useState<string | undefined>(undefined);
-
-    const predictedFieldName = getPredictedFieldName(
-      jobConfig.dest.results_field,
-      jobConfig.analysis
-    );
-
-    const dependentVariable = getDependentVar(jobConfig.analysis);
-
-    function toggleColumnsPopover() {
-      setColumnsPopoverVisible(!isColumnsPopoverVisible);
-    }
-
-    function closeColumnsPopover() {
-      setColumnsPopoverVisible(false);
-    }
-
-    function toggleColumn(column: EsFieldName) {
-      if (tableItems.length > 0 && jobConfig !== undefined) {
-        // spread to a new array otherwise the component wouldn't re-render
-        setSelectedFields([
-          ...toggleSelectedField(selectedFields, column, jobConfig.dest.results_field, depVarType),
-        ]);
-      }
-    }
-
     const needsDestIndexFields = indexPattern && indexPattern.title === jobConfig.source.index[0];
-
+    const resultsField = jobConfig.dest.results_field;
     const {
       errorMessage,
-      loadExploreData,
-      sortField,
-      sortDirection,
-      status,
-      tableItems,
-    } = useExploreData(
-      jobConfig,
-      needsDestIndexFields,
+      fieldTypes,
+      pagination,
+      searchQuery,
       selectedFields,
+      rowCount,
+      setPagination,
+      setSearchQuery,
       setSelectedFields,
-      setDocFields,
-      setDepVarType
-    );
+      setSortingColumns,
+      sortingColumns,
+      status,
+      tableFields,
+      tableItems,
+    } = useExploreData(jobConfig, needsDestIndexFields);
 
-    const columns: Array<ColumnType<TableItem>> = selectedFields
-      .sort(({ name: a }, { name: b }) => sortRegressionResultsFields(a, b, jobConfig))
-      .map(field => {
-        const { type } = field;
-        let format: any;
+    useEffect(() => {
+      setEvaluateSearchQuery(searchQuery);
+    }, [JSON.stringify(searchQuery)]);
 
-        if (indexPattern !== undefined) {
-          format = mlFieldFormatService.getFieldFormatFromIndexPattern(indexPattern, field.id, '');
-        }
+    const columns = tableFields
+      .sort((a: any, b: any) => sortRegressionResultsFields(a, b, jobConfig))
```

Comment:
Should this be a dedicated `sortClassificationResultsFields()` function? If it works as expected and it's just a naming issue than I can do that in the follow-up clean-up I'm working on in #63447

Reply:
Ah, good find. That function is actually used to sort both classification and regression fields so it needs to be renamed to something more generic like `sortAnalyticsResultsFields` or something like that. It's just a naming issue - sounds good to update in the follow-up you've got in progress. Thank you 👍